jobspec2/types.config.go

```diff
@@ -308,8 +309,26 @@ func (c *jobConfig) EvalContext() *hcl.EvalContext {
 			inputVariablesAccessor: cty.ObjectVal(vars),
 			localsAccessor:         cty.ObjectVal(locals),
 		},
-		UnknownVariable: func(expr string) (cty.Value, error) {
-			v := "${" + expr + "}"
+		UndefinedVariable: func(t hcl.Traversal) (cty.Value, hcl.Diagnostics) {
+			body := c.ParseConfig.Body
+			start := t.SourceRange().Start.Byte
```

Comment:
Just for my own clarity here, because the the docs on `hcl.Traversal` aren't very easy to drop into and come away with a solid understanding... the `SourceRange` starts right after the `${` we're evaluating and stops either at the `}` or the end of the body? Just want to make sure that's guaranteed so that the slicing below can't panic.

Reply:
Imperially, Start.Byte is the offset first non-white space character after `${`, and End.Byte is the offset of the first whitespace or `}`.

It's an error if `${` is present without a corresponding `}`.  I was being a bit more defensive though.